server/events/project_command_builder.go

```diff
@@ -178,7 +178,21 @@ func (p *DefaultProjectCommandBuilder) buildPlanAllCommands(ctx *CommandContext,
 			// the actual clone directory.
 		}
 	}
-
+	pullDir, err := p.WorkingDir.GetPullDir(ctx.Pull.BaseRepo, ctx.Pull)
+	if err != nil {
+		return nil, err
+	}
+	// Check if the PR has been already locked and if so drop all pending plans
+	locked, err := p.WorkingDirLocker.IsPullLocked(ctx.Pull.BaseRepo.FullName, ctx.Pull.Num)
+	if err != nil {
+		return nil, err
+	}
+	if locked {
+		errDeletePlans := p.PendingPlanFinder.DeletePlans(pullDir)
+		if errDeletePlans != nil {
+			return nil, errDeletePlans
+		}
```

Comment:
We probably want to acquire a lock on the workdir before doing this operation, to make sure we're not deleting plans while atlantis is planning/applying.

If we execute this later in the method between aquiring the workspace lock and cloning the repo it should do what we expect.